File: java/src/main/java/com/experian/dvs/client/address/format/components/AdditionalLocality.java
package com.experian.dvs.client.address.format.components;

import com.experian.dvs.client.server.address.format.RestApiAddressFormatComponents;

public class AdditionalLocality {

    private final AdditionalSubRegion subRegion;

    // Constructor
    public AdditionalLocality(RestApiAddressFormatComponents.RestApiAddressComponentAdditionalLocality locality) {
        this.subRegion = locality.getSubRegion() != null ? new AdditionalSubRegion(locality.getSubRegion()) : null;
    }

    // Getters
    public AdditionalSubRegion getSubRegion() {
        return subRegion;
    }
}

File: csharp/DVSClient/DVSClient/address/format/components/AdditionalLocality.cs
using DVSClient.Server.Address.Format;

namespace DVSClient.Address.Format.Components
{
    public class AdditionalLocality
    {
        public AdditionalSubRegion? SubRegion { get; }

        public AdditionalLocality(RestApiAddressFormatComponents.RestApiAddressComponentAdditionalLocality api)
        {
            SubRegion = api.SubRegion != null ? new AdditionalSubRegion(api.SubRegion) : null;
        }
    }
}